src/components/pages/dashboard-pages/EditProfileForm.jsx

```diff
@@ -1,36 +1,113 @@
 
-const EditProfileForm = ({ options, role, cancel }) => {
-  // this is just a layout without sending the data to DB and storing the state
+const EditProfileForm = ({
+  options,
+  role,
+  cancel,
+  formData,
+  onChange,
+  onHandleSubjects,
+  category,
+  onSubmit,
+  formErrors,
+  userPhotoSrc,
+  onPhotoChange,
+  onPhotoSubmit,
+  isUpload,
+  setIsUpload,
+}) => {
   return (
     <div className="w-full h-full md:h-screen flex flex-grow items-center sm:items-start sm:justify-center sm:flex-row flex-col overflow-y-hidden">
-      <div className="sm:w-1/5 w-4/5 flex flex-col items-center gap-4 mt-4 p-4">
-        {/* div for user's photo */}
-        <div className="h-24 w-24 bg-grey rounded-full"></div>
-        <button className="bg-red hover:bg-pureWhite hover:text-red h-8 w-1/3 sm:w-2/4 hover:border-2 hover:border-red text-white font-spartan font-semibold text-base sm:text-lg rounded-lg transition duration-300 easy-in">
-          Add Photo
-        </button>
+      <div className="sm:w-1/4 w-4/5 flex flex-col items-center gap-4 mt-4 p-4">
+        <img
+          className="h-24 w-24 rounded-full"
+          src={userPhotoSrc}
+          alt="user photo"
+        />
+        {!isUpload && (
+          <button
+            className="bg-red hover:bg-pureWhite hover:text-red h-8 w-1/3 sm:w-3/4 md:w-1/2 hover:border-2 hover:border-red text-white font-spartan font-semibold text-sm sm:text-lg rounded-lg transition duration-300 easy-in"
+            onClick={() => setIsUpload(!isUpload)}
+          >
+            Add Photo
+          </button>
+        )}
+        {isUpload && (
+          <>
+            <form
+              onSubmit={onPhotoSubmit}
+              className="flex flex-col items-center"
+            >
+              <FormInput type="file" onChange={onPhotoChange} />
+              <button
+                type="submit"
+                className="bg-red hover:bg-pureWhite hover:text-red h-8 w-1/2 sm:w-3/4 md:w-1/2 hover:border-2 hover:border-red text-white font-spartan font-semibold text-sm sm:text-lg rounded-lg transition duration-300 easy-in"
+              >
+                Upload
+              </button>
+            </form>
+          </>
+        )}
       </div>
-      <form className="mt-4 p-4 sm:w-2/5 w-4/5 h-full">
+      <form className="mt-4 p-4 sm:w-2/5 w-4/5 h-full" onSubmit={onSubmit}>
+        {formErrors.firstName && (
+          <p className="text-red text-sm font-spartan">
+            {formErrors.firstName}
+          </p>
+        )}
+        {formErrors.lastName && (
+          <p className="text-red text-sm font-spartan">{formErrors.lastName}</p>
+        )}
+        {formErrors.form && (
+          <p className="text-red text-sm font-spartan">{formErrors.form}</p>
+        )}
         <div className="flex gap-8">
-          <FormInput placeholder=" " name="firstName">
+          <FormInput
+            placeholder=" "
+            name="firstName"
```

Comment:
There are some inconsistencies with the casing of the value throughout some forms. Some use "camelCase" others "kebab-case".  If this is not deliberate I recommend consistency. This also means that you need to anticipate what the backend is expecting.

Reply:
As I understand kebab case is used by tailwind by default.
We use camelCase for attributes, names of the functions  and so on as the backend team does. For components we use PascalCase.

Reply:
For now as long as the backend is receiving the values in the expected casing this will be fine for now.